src/wp-includes/html-api/class-wp-html-doctype-info.php

```diff
@@ -0,0 +1,269 @@
+<?php
+/**
+ * HTML API: WP_HTML_Doctype_Info class
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ * @since 6.7.0
+ */
+
+/**
+ * Core class used by the HTML API processor to represent a DOCTYPE declaration.
+ *
+ * @since 6.7.0
+ */
+class WP_HTML_Doctype_Info {
+	/**
+	 * The name of the DOCTYPE.
+	 *
+	 * @var string|null
+	 */
+	private $name;
+
+	/**
+	 * The public identifier of the DOCTYPE.
+	 *
+	 * @var string|null
+	 */
+	private $public_identifier;
+
+	/**
+	 * The system identifier of the DOCTYPE.
+	 *
```

Comment:
I'm just guessing but I bet a lot could be added to these properties and this class to educate people on the value (or danger) of these parts of the DOCTYPE. Just a comment saying something like "This exists mostly to be able to represent and copy HTML from one document from another, but it's best to use only the standard `<!DOCTYPE html>` declaration for HTML."

just doodling